src/main/java/com/powsybl/network/conversion/server/NetworkStreamingService.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.network.conversion.server;
+
+import com.powsybl.commons.datasource.DirectoryDataSource;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.network.conversion.server.dto.ExportNetworkInfos;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Comparator;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * @author Rehili Ghazwa <ghazwa.rehili at rte-france.com>
+ */
+
+@Service
+public class NetworkStreamingService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkStreamingService.class);
+    private static final String ZIP_EXTENSION = ".zip";
+    private static final String TEMP_DIR_PREFIX = "network-export-";
+
+    public ExportNetworkInfos getExportNetworkInfos(Network network, String format,
+                                                    String fileOrNetworkName, Properties exportProperties,
+                                                    long networkSize) {
+        return withTempDirectory(tempDir -> {
+            Path finalFile = exportNetworkFiles(network, format, fileOrNetworkName, exportProperties, tempDir);
+            long fileSize = getFileSize(finalFile);
+            return new ExportNetworkInfos(fileOrNetworkName + "." + format.toLowerCase(), finalFile, networkSize, fileSize);
+        });
+    }
+
+    public Path streamNetworkToFile(Network network, String format, Properties exportProperties, String fileName) {
+        return withTempDirectory(tempDir -> exportNetworkFiles(network, format, fileName, exportProperties, tempDir));
+    }
+
+    private Path exportNetworkFiles(Network network, String format, String fileName, Properties exportProperties, Path tempDir) {
+        try {
+            DirectoryDataSource dataSource = new DirectoryDataSource(tempDir, fileName);
+            network.write(format, exportProperties, dataSource);
```

Comment:
Here you write the file in /tmp. Why do you write it again at row 63 ? You can make it simpler